src/exportAuthRecord.ts

```diff
@@ -0,0 +1,143 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { callWithTelemetryAndErrorHandling, IActionContext, registerEvent } from '@microsoft/vscode-azext-utils';
+import * as fs from 'fs-extra';
+import * as os from 'os';
+import * as path from 'path';
+import type { ExtensionContext } from 'vscode';
+import * as vscode from 'vscode';
+import { ext } from './extensionVariables';
+
+/**
+ * Registers the exportAuthRecord callback for session changes and ensures the auth record is exported at least once on activation.
+ */
+export function registerExportAuthRecordOnSessionChange(_context: ExtensionContext) {
+    registerEvent(
+        'treeView.onDidChangeSessions',
+        (vscode as typeof import('vscode')).authentication.onDidChangeSessions,
+        exportAuthRecord
+    );
+
+    // Also call exportAuthRecord once on activation to ensure the auth record is exported at least once
+    // (onDidChangeSessions is not guaranteed to fire on activation)
+    void callWithTelemetryAndErrorHandling('azureResourceGroups.exportAuthRecord', async (actionContext) => {
+        await exportAuthRecord(actionContext);
+    });
+}
+
+/**
+ * Exports the current authentication record to a well-known location in the user's .azure directory.
+ * Used for interoperability with other tools and applications.
+ */
+export async function exportAuthRecord(context: IActionContext): Promise<void> {
+    const AUTH_PROVIDER_ID = 'microsoft'; // VS Code Azure auth provider
+    const SCOPES = ['https://management.azure.com/.default']; // Default ARM scope
+
+    context.errorHandling.suppressDisplay = true;
+    context.telemetry.suppressIfSuccessful = true;
```

Comment:
If we want to capture telemetry on success as well as failure, so we can gauge the success rate, we will need to remove this line. However, I am worried it might be too noisy of an event--`onDidChangeSessions` gets called quite a bit, sometimes in rapid succession. Anyone have thoughts?